### PR TITLE
Use Remix template eslint config

### PR DESCRIPTION
**Pull request by sebastien-comeau**
### Related Azure Boards Work Items

Use Remix template eslint config and fix eslit errors. `@remix-run/eslint` config is deprecated in v2.

https://github.com/remix-run/remix/issues/7056

### Checklist

- [x] I have tested the changes locally
- [x] I have updated the documentation if necessary
- [x] I have added/updated tests that prove my fix is effective or that my feature works
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`
- [x] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [x] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: frontend/.eslintrc.cjs
Comment (gregory-j-baker) [inline frontend/.eslintrc.cjs:1]: This can't be an esm?

File: frontend/.eslintrc.cjs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+ * This is intended to be a basic starting point for linting in your app.
+ * It relies on recommended configs out of the box for simplicity, but you can
+ * and should modify this configuration to best suit your team's needs.
+ */
+
+/** @type {import('eslint').Linter.Config} */
+module.exports = {
+  root: true,
+  parserOptions: {
+    ecmaVersion: 'latest',
```
Comment (gregory-j-baker) [inline frontend/.eslintrc.cjs:11]: This should match what's in `tsconfig.json`

File: frontend/.eslintrc.cjs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    sourceType: 'module',
+    ecmaFeatures: {
+      jsx: true,
+    },
+  },
+  env: {
+    browser: true,
+    commonjs: true,
```
Comment (gregory-j-baker) [inline frontend/.eslintrc.cjs:19]: We aren't using commonjs

**Review by gregory-j-baker — CHANGES_REQUESTED** [review]

File: frontend/.eslintrc.cjs (hunk 2 above)
Comment (sebastien-comeau) [reply frontend/.eslintrc.cjs:19]: It give me the following error if I remove it or set it to `false`.

image

File: frontend/.eslintrc.cjs (hunk 1 above)
Comment (sebastien-comeau) [reply frontend/.eslintrc.cjs]: Changed it to 2022.

File: frontend/.eslintrc.cjs (hunk 2 above)
Comment (gregory-j-baker) [reply frontend/.eslintrc.cjs:19]: So we have to enable it so eslint can lint eslint. 🤣

**Review by gregory-j-baker — APPROVED** [review]

File: frontend/.eslintrc.cjs (hunk 2 above)
Comment (sebastien-comeau) [reply frontend/.eslintrc.cjs:19]: I guess it's needed. I tried and failed. 🤷

**Pull request merged**